src/libraries/System.Text.Json/src/System/Text/Json/Serialization/DefaultReferenceResolver.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the.NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Text.Json
+{
+    internal sealed class DefaultReferenceResolver
+    {
+        private uint _referenceCount;
+        private Dictionary<string, object> _keyObjectMap;
+        private Dictionary<object, string> _objectKeyMap;
+
+        public DefaultReferenceResolver()
+        {
+            _keyObjectMap = new Dictionary<string, object>();
+            _objectKeyMap = new Dictionary<object, string>(ReferenceEqualsEqualityComparer<object>.Comparer);
+        }
+
+        // Used on deserialization.
```

Comment:
Please rename in a way that you won't need this comment and similar instances below.